src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -2596,5 +2596,201 @@ .maxstack  1
+                Console.WriteLine(""Euint[]"");
+                break;
+
+            case int[][] _ when o.GetType() == typeof(int[][]):
+                Console.WriteLine(""int[][]"");
+                break;
+            case uint[][] _ when o.GetType() == typeof(uint[][]):
+                Console.WriteLine(""uint[][]"");
+                break;
+            case Eint[][] _ when o.GetType() == typeof(Eint[][]):
+                Console.WriteLine(""Eint[][]"");
+                break;
+            case Euint[][] _ when o.GetType() == typeof(Euint[][]):
+                Console.WriteLine(""Euint[][]"");
+                break;
+
+            case long[] _ when o.GetType() == typeof(long[]):
+                Console.WriteLine(""long[]"");
+                break;
+            case ulong[] _ when o.GetType() == typeof(ulong[]):
+                Console.WriteLine(""ulong[]"");
+                break;
+            case Elong[] _ when o.GetType() == typeof(Elong[]):
+                Console.WriteLine(""Elong[]"");
+                break;
+            case Eulong[] _ when o.GetType() == typeof(Eulong[]):
+                Console.WriteLine(""Eulong[]"");
+                break;
+
+            case IntPtr[] _ when o.GetType() == typeof(IntPtr[]):
+                Console.WriteLine(""IntPtr[]"");
+                break;
+            case UIntPtr[] _ when o.GetType() == typeof(UIntPtr[]):
+                Console.WriteLine(""UIntPtr[]"");
+                break;
+
+            case IntPtr[][] _ when o.GetType() == typeof(IntPtr[][]):
+                Console.WriteLine(""IntPtr[][]"");
+                break;
+            case UIntPtr[][] _ when o.GetType() == typeof(UIntPtr[][]):
+                Console.WriteLine(""UIntPtr[][]"");
+                break;
+
+            default:
+                Console.WriteLine(""oops: "" + o.GetType());
+                break;
+        }
+    }
+}
+enum Ebyte : byte {}
+enum Esbyte : sbyte {}
+enum Eshort : short {}
+enum Eushort : ushort {}
+enum Eint : int {}
+enum Euint : uint {}
+enum Elong : long {}
+enum Eulong : ulong {}
+";
+            var expectedOutput =
+@"byte[]
+sbyte[]
+Ebyte[]
+Esbyte[]
+short[]
+ushort[]
+Eshort[]
+Eushort[]
+int[]
+uint[]
+Eint[]
+Euint[]
+int[][]
+uint[][]
+Eint[][]
+Euint[][]
+long[]
+ulong[]
+Elong[]
+Eulong[]
+IntPtr[]
+UIntPtr[]
+IntPtr[][]
+UIntPtr[][]
+";
+            var compilation = CreateCompilation(source, options: TestOptions.ReleaseExe);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: expectedOutput);
+            compilation = CreateCompilation(source, options: TestOptions.DebugExe);
+            compilation.VerifyDiagnostics();
+            CompileAndVerify(compilation, expectedOutput: expectedOutput);
+        }
+
+        [Fact]
+        public void CompileTimeRuntimeInstanceofMismatch_02()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        object o = null;
```

Comment:
>object o = null; [](start = 8, length = 16)

Perhaps change the code to test both `byte[]` and `sbyte[]` and report which branch was hit.